platform/saivpp/vpplib/SwitchStateBaseAcl.cpp

```diff
@@ -405,6 +405,148 @@ sai_status_t acl_rule_field_update(
+    {
+        rif_oid = attr.value.oid;
+    } else {
+        SWSS_LOG_ERROR("RIF missing in nexthop %s", sai_serialize_object_id(next_hop_oid).c_str());
+        return SAI_STATUS_SUCCESS;
+    }
+
+    attr.id = SAI_ROUTER_INTERFACE_ATTR_PORT_ID;
+    if (get(SAI_OBJECT_TYPE_ROUTER_INTERFACE, rif_oid, 1, &attr) == SAI_STATUS_SUCCESS)
+    {
+        port_oid = attr.value.oid;
+    } else {
+        SWSS_LOG_ERROR("Port ID is missing missing in RIF object %s", sai_serialize_object_id(rif_oid).c_str());
+        return SAI_STATUS_SUCCESS;
+    }
+
+    attr.id = SAI_ROUTER_INTERFACE_ATTR_OUTER_VLAN_ID;
+    if (get(SAI_OBJECT_TYPE_ROUTER_INTERFACE, rif_oid, 1, &attr) == SAI_STATUS_SUCCESS)
+    {
+        vlan_id = attr.value.u16;
+    }
+
+    std::string hwif_name;
+    if (!vpp_get_hwif_name(port_oid, vlan_id, hwif_name)) {
+        SWSS_LOG_WARN("VPP hwif name not found for port %s", sai_serialize_object_id(port_oid).c_str());
+        return SAI_STATUS_SUCCESS;
+    }
+
+    switch (ip_address.addr_family) {
+        case SAI_IP_ADDR_FAMILY_IPV4:
+        {
+            sai_ip_address_t_to_vpp_ip_addr_t(ip_address, rule->next_hop_ip);
+            rule->ip_protocol = 1; // IP46_TYPE_IP4=1, IP46_TYPE_IP6=2
+            break;
+        }
+        case SAI_IP_ADDR_FAMILY_IPV6:
+        {
+            sai_ip_address_t_to_vpp_ip_addr_t(ip_address, rule->next_hop_ip);
+            rule->ip_protocol = 2; // IP46_TYPE_IP4=1, IP46_TYPE_IP6=2 
+            break;
+        }
+        default:
+            break;
+    }
+
+    strncpy(rule->hwif_name, hwif_name.c_str(), sizeof(rule->hwif_name) -1);
+
+    SWSS_LOG_NOTICE("Tunterm rule received: IP Protocol %d, rif_oid %d, next-hop hwif_name %s", rule->ip_protocol,
+                    rif_oid, rule->hwif_name);
+
+    vpp_ip_addr_t_to_string(&rule->next_hop_ip, nh_ip_str, INET6_ADDRSTRLEN);        
+    SWSS_LOG_NOTICE("Tunterm acl rule has next-hop IP %s", nh_ip_str);    
+
+    return status;
+}
+
+sai_status_t SwitchStateBase::tunterm_acl_rule_field_update(
+    _In_ sai_acl_entry_attr_t          attr_id,
+    _In_ const sai_attribute_value_t  *value,
+    _Out_ tunterm_acl_rule_t      *rule)
+{
+    sai_status_t status;
+    char         dst_ip_str[INET6_ADDRSTRLEN];
+
+    assert(NULL != value);
+    status = SAI_STATUS_SUCCESS;
+
+    switch (attr_id) {
+    case SAI_ACL_ENTRY_ATTR_FIELD_DST_IP:
+    case SAI_ACL_ENTRY_ATTR_FIELD_INNER_DST_IP:
+    {
+        vpp_ip_addr_t *ip_addr, *ip_mask;
+        ip_addr = &rule->dst_prefix;
+        ip_mask = &rule->dst_prefix_mask;
+        struct sockaddr_in *sin =  &ip_addr->addr.ip4;
+        ip_addr->sa_family = AF_INET;
+        sin->sin_addr.s_addr = value->aclfield.data.ip4;
+        sin =  &ip_mask->addr.ip4;
+        sin->sin_addr.s_addr = value->aclfield.mask.ip4;
+        vpp_ip_addr_t_to_string(ip_addr, dst_ip_str, INET6_ADDRSTRLEN);
+        SWSS_LOG_NOTICE("Tunterm acl rule has dst IP: %s", dst_ip_str);
```

Comment:
Please change all NOTICE to INFO except very few important one like tunterm acl created/modify/deleted successfully. See issue #116 .

Reply:
Updated